test/jdk/javax/swing/ImageIcon/LoadInterruptTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.MediaTracker;
+
+import javax.swing.ImageIcon;
+import javax.swing.SwingUtilities;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+/*
+ * @test
+ * @bug 8236987
+ * @summary Verifies ImageIcon constructor produces no output when the
+ *          thread is interrupted
+ * @run main LoadInterruptTest
+ */
+
+public class LoadInterruptTest {
+    private static ByteArrayOutputStream testOut;
+    private static PrintStream prevSysOut;
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            SwingUtilities.invokeAndWait(new Runnable() {
+                public void run() {
+                    loadImageIcon();
+                }
+            });
+        } finally {
+            unsetOutput();
+        }
+    }
+
+    public static void setUpOutput() {
+        prevSysOut = System.out;
+        testOut = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(testOut, true, StandardCharsets.UTF_8));
```

Comment:
The spec says "First, if there is a security manager, its checkPermission method is called with a RuntimePermission("setIO") permission to see if it's ok to reassign the "standard" output stream."

Since security manager is removed, I guess either this doc needs updation or we still need to follow this checkPermission step...

Reply:
What are referring to?

The javadoc to [`System.setOut`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/System.html#setOut(java.io.PrintStream)) does indeed state this. [The implementation](https://github.com/openjdk/jdk/blob/4dbebb62aa264adda8d96a06f608ef9d13155a26/src/java.base/share/classes/java/lang/System.java#L244) does exactly what it says: it calls `checkIO` method which verifies whether `"setIO"` permission is granted if `SecurityManager` is installed.

I can't see what we should do about it. `SecurityManager` hasn't been removed yet.

Reply:
Yeah, right..I was thinking something else..